alignak/objects/service.py

```diff
@@ -923,65 +928,60 @@ def duplicate(self, host):
-                                                                              key_value[key]))
-                    if hasattr(self, 'service_dependencies'):
-                        for i, servicedep in enumerate(new_s.service_dependencies):
-                            new_s.service_dependencies[i] = servicedep.replace(
-                                '$' + key + '$', key_value[key]
-                            )
-                # And then add in our list this new service
-                duplicates.append(new_s)
-        else:
-            # If error, we should link the error to the host, because self is
-            # a template, and so won't be checked not print!
-            if errcode == GET_KEY_VALUE_SEQUENCE_ERROR_SYNTAX:
-                err = "The custom property '%s' of the host '%s'" \
-                      "is not a valid entry %s for a service generator" % \
-                      (self.duplicate_foreach.strip(), host.get_name(), entry)
-                logger.warning(err)
-                host.configuration_errors.append(err)
-            elif errcode == GET_KEY_VALUE_SEQUENCE_ERROR_NODEFAULT:
-                err = "The custom property '%s 'of the host '%s' has empty " \
-                      "values %s but the service %s has no default_value" % \
-                      (self.duplicate_foreach.strip(),
-                       host.get_name(), entry, self.service_description)
-                logger.warning(err)
-                host.configuration_errors.append(err)
-            elif errcode == GET_KEY_VALUE_SEQUENCE_ERROR_NODE:
-                err = "The custom property '%s' of the host '%s' has an invalid node range %s" % \
-                      (self.duplicate_foreach.strip(), host.get_name(), entry)
-                logger.warning(err)
-                host.configuration_errors.append(err)
+        try:
+            key_values = tuple(generate_key_value_sequences(entry, default_value))
+        except KeyValueSyntaxError as exc:
+            reason = "is not a valid entry for a service generator: %(exc)s"
+
+            fmt_dict = {
+                'prop': self.duplicate_foreach,
+                'host': host.get_name(),
+                'svc': self.service_description,
+                'entry': entry,
+                'exc': exc,
+                'reason': reason,
+            }
+            err = (
+                "The custom property %(prop)r of the "
+                "host %(host)r %(reason)s, with entry=%(entry)r") % fmt_dict
+            logger.warning(err)
+            host.configuration_errors.append(err)
+            return duplicates
+
+        for key_value in key_values:
+            key = key_value['KEY']
+            # Maybe this key is in the NOT list, if so, skip it
+            if key in not_keys:
+                continue
+            new_s = self.copy()
+            new_s.host_name = host.get_name()
+            if self.is_tpl():  # if template, the new one is not
+                new_s.register = 1
```

Comment:
as it's a bool prop i would put a bool here. Strange that there was no bug before...